ldap/servers/slapd/back-ldbm/archive.c

```diff
@@ -490,3 +490,212 @@ ldbm_back_ldbm2archive(Slapi_PBlock *pb)
     slapi_ch_free_string(&directory);
     return return_value;
 }
+
+#define CPRETRY 4
+/* Copy a file to the backup */
+static int32_t
+archive_copyfile(char *source, char *destination, char *filename, mode_t mode, Slapi_Task *task)
+{
+    int source_fd = -1;
+    int dest_fd = -1;
+    char *buffer = NULL;
+    int return_value = -1;
+    int bytes_to_write = 0;
+    char *dest_file = NULL;
+
+    if (access(source, F_OK) != 0) {
```

Comment:
You may also use PR_Access/PR_Open/PR_Read/PR_Write/... ?

Reply:
> You may also use PR_Access/PR_Open/PR_Read/PR_Write/... ?

Fixed